packages/react/src/utils/renderComponent.tsx

```diff
@@ -109,12 +109,18 @@ const renderComponent = <P extends {}>(
     : {}
 
   // Resolve styles using resolved variables, merge results, allow props.styles to override
-  const mergedStyles: ComponentSlotStylesPrepared = mergeComponentStyles(
-    theme.componentStyles[displayName],
-    withDebugId({ root: props.design }, 'props.design'),
-    withDebugId({ root: props.styles }, 'props.styles'),
-    withDebugId({ root: animationCSSProp }, 'props.animation'),
-  )
+  let mergedStyles: ComponentSlotStylesPrepared = theme.componentStyles[displayName] || {
```

Comment:
As the theme is immutable, the mergedStyles by default will always reference to the same object.